remove rephraser

lyie28 opened the pull request.

adeprez commented:
Ok for the changes. The rephraser is still used in the evaluator and in the logger.ipynb notebook, can you remove it there too?

We had a `rephrase_dataset` that is no longer relevant

lyie28 commented:
@adeprez Oki, I changed it so that the retriever and llm evaluators look for "query" in the dataframe instead of "retriever_query" and "llm_query" which were created by the rephrase_dataset method.

I also needed to update all the docs for eval + I also removed all driver arguments to the bm25 in these since it no longer accepts a driver argument.
